sonar-db/src/main/java/org/sonar/db/version/DropTableBuilder.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.version;
+
+import java.util.List;
+import java.util.stream.Stream;
+import org.sonar.core.util.stream.Collectors;
+import org.sonar.db.dialect.Dialect;
+import org.sonar.db.dialect.Oracle;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Stream.of;
+import static org.sonar.db.version.Validations.TABLE_NAME_MAX_SIZE;
+import static org.sonar.db.version.Validations.checkDbIdentifier;
+
+public class DropTableBuilder {
```

Comment:
there are some differences with Ruby ActiveRecord layer:
* I think table name must be lower-cased when generating names of sequence and trigger. Generally I would also lower-case table name should be enforced, as it's done in `CreateTableBuilder` with `checkDbIdentifier()`.
* on Oracle, if sequence can not be dropped then error is ignored (probably because the sequence may not exist). See `rescue nil` at https://github.com/SonarSource/sonarqube/blob/52fec9954c6d5225d6ea4ee61a3ae71721fd3881/server/sonar-web/src/main/webapp/WEB-INF/gems/gems/activerecord-jdbc-adapter-1.1.3/lib/arjdbc/oracle/adapter.rb#L145-L145.
* on MSSQL table name is surrounded with square brackets : https://github.com/SonarSource/sonarqube/blob/8813433bbee5e54f99224e3fb072019e894ae7d3/server/sonar-web/src/main/webapp/WEB-INF/gems/gems/activerecord-jdbc-adapter-1.1.3/lib/arjdbc/mssql/adapter.rb#L224-L230